app/src/main/java/ch/sdp/vibester/activity/GamescreenActivity.kt

```diff
@@ -18,25 +18,25 @@ class GamescreenActivity: AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_gamescreen)
 
-        /* number of players for the game
-            currently hardcoded as a placeholder
-            will be retrieved from intent launched from the game setup screen
-         */
-        val players = arrayOf("Kamila", "Jiabao", "Arda", "Laurynas")
+        val getIntent = intent.extras
+        val nPlayers = getIntent?.getInt("Number of players")
 
         val answer = findViewById<LinearLayout>(R.id.answer)
         val answerText = findViewById<TextView>(R.id.answerText)
+        answerText.text= "The song was Demo by The Placeholders"
 
-        // hardcoded test values
-        val song = "Demo"
-        val artist = "The Placeholders"
+        val allPoints = nPlayers?.let { Array<Int>(it, { i -> 0 }) }
 
-        answerText.text= "The song was $song by $artist"
+        val playersFull = getIntent?.getStringArray("Player Names")
+        val players = nPlayers?.let { playersFull?.copyOfRange(0, it) }
 
-        val allPoints = arrayOf(1, 2, 3, 4)
+        println(nPlayers)
+        print(players) // test
 
-        buildScores(players, allPoints)
-        buildBuzzers(players, answer)
+        val buzIds = players?.let { fetchBuzIdArray(it.size) }
+
+        if (players != null && allPoints != null) { buildScores(players, allPoints) }

```

Comment:
for buildScore function: I am not sure if you need allPoints here, since it is initialization and it is always 0